tests/integration/test_hive_query/test.py

```diff
@@ -0,0 +1,91 @@
+import logging
+import os
+
+import pytest
+from helpers.cluster import ClickHouseCluster
+from helpers.test_tools import TSV
+
+logging.getLogger().setLevel(logging.INFO)
+logging.getLogger().addHandler(logging.StreamHandler())
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+
+
+@pytest.fixture(scope="module")
+def started_cluster():
```

Comment:
Also test for caching? You can use profile events `ExternalDataSourceLocalCacheReadBytes` and check that data was actually read from cache.

Reply:
Is there any way to read to ProfileEvent from client?

Reply:
done